### PR TITLE
Qianlin-Qin Week 4

### PR DESCRIPTION
**Pull request by AriaQin210**

### CODE REVIEW

File: hw04/BarbieWalletBalance.cs [hunk 1]
```diff
@@ -13,24 +13,51 @@ public class BarbieBank : MonoBehaviour
     BarbieBank should take in the parameters: numOfPennies, cashAmount, and numOfCreditCards
     Have the function return a new generic array with the correct parameters. 
     */
-
     void Start()
     {
+
```
Comment (debbieyuen) [inline hw04/BarbieWalletBalance.cs:24]: We will need to inherit from `BarbieWorld.cs`. We write the name of the class and then the name of the method followed by the brackets. 

```
GenericClass<int> myClass = new GenericClass<int>();
```

or for this problem:

```
BarbieWorld<string> barbieWorld = new BarbieWorld<string>();
barbieWorld.BarbieCareers("FashionModel");
```

File: hw04/BarbieWalletBalance.cs [hunk 2]
```diff
@@ -13,24 +13,51 @@ public class BarbieBank : MonoBehaviour
     BarbieBank should take in the parameters: numOfPennies, cashAmount, and numOfCreditCards
     Have the function return a new generic array with the correct parameters. 
     */
-
     void Start()
     {
+        
         int[] biggerWallet = BarbieWallet(500, 600);
```
Comment (debbieyuen) [inline hw04/BarbieWalletBalance.cs:25]: We need to modify the code here to take in generics 
```
    void Start()
    {
       // Convert function to take in generic values
        int[] biggerWallet = BarbieWallet<int>(500, 600);
        Debug.Log(biggerWallet.Length + " " + biggerWallet[0] + " " + biggerWallet[1]);

        GetMoney<int>(500, 600);
    }
```

File: hw04/BarbieWalletBalance.cs [hunk 3]
```diff
@@ -13,24 +13,51 @@ public class BarbieBank : MonoBehaviour
...
         Debug.Log(biggerWallet.Length + " " + biggerWallet[0] + " " + biggerWallet[1]);
 
         GetMoney(500, 600);
     }
+    private T[] BarbieWallet<T>(T money1, T money2)
```
Comment (debbieyuen) [inline hw04/BarbieWalletBalance.cs:30]: good :)

File: hw04/BarbiesHouse.cs [hunk 4]
```diff
@@ -10,5 +10,20 @@ BarbieWorld and write at least three methods within BarbieHouse representing
 */ 
 public class BarbieHouse : MonoBehaviour
```
Comment (debbieyuen) [inline hw04/BarbiesHouse.cs:11]: This part is incorrect. 

```
public class BarbieHouse : BarbieWorld
{
     public void DisplayFurniture()
     {
          Debug.Log("Barbie has a couch and a bed.");
     }

     public void ShowPets()
     {
          Debug.Log("Barbie has a dog.");
     }

     public void ShowFood()
     {
          Debug.Log("Barbie's fridge is stocked with salads.");
     }
}

**Review by debbieyuen — COMMENTED** [review]
thank you Aria!!

**Comment by AriaQin210** [comment]
Thank you so much!!╰(*°▽°*)╯

**Pull request closed without merging**